packages/angular/cli/commands/update-impl.ts

```diff
@@ -879,22 +857,63 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
   }
 
   /**
-   * Checks if the current installed CLI version is older than the latest version.
-   * @returns `true` when the installed version is older.
+   * Checks if the current installed CLI version is older or newer than a compatible version.
+   * @returns the version to install or null when there is no update to install.
    */
-  private async checkCLILatestVersion(verbose = false, next = false): Promise<boolean> {
-    const installedCLIVersion = VERSION.full;
-
-    const LatestCLIManifest = await fetchPackageManifest(
-      `@angular/cli@${next ? 'next' : 'latest'}`,
-      this.logger,
-      {
-        verbose,
-        usingYarn: this.packageManager === PackageManager.Yarn,
-      },
-    );
+  async checkCLIVersion(
+    packagesToUpdate: string[] | undefined,
+    verbose = false,
+    next = false,
+  ): Promise<string | null> {
+    let cliRunnerTargetVersion: string | number;
```

Comment:
Could this be improved with:

```typescript
const cliRunnerTargetVersion = getRunnerTargetVersion(/* ... */);

function getRunnerTargetVersion(/* ... */): string { /* ... */ }
```

Then all that logic can be encapsulated in its own function and simplified with `return`.